src/pudl_rmi/connect_deprish_to_ferc1.py

```diff
@@ -650,3 +652,141 @@ def _allocate_col(
+        df1=df1, df2=df2, data_col='plant_balance_w_common')
+
+    # known baddies. We need to track these down
+    assert(len(plant_bad) <= 53)
+    assert(len(util_bad) <= 88)
+
+    util_bad, plant_bad = data_col_test(
+        df1=df1, df2=df2, data_col='unaccrued_balance_w_common')
+
+    # known baddies. We need to track these down
+    assert(len(plant_bad) <= 35)
+    assert(len(util_bad) <= 70)
+
+
+def data_col_test(df1, df2, data_col: str) -> pd.DataFrame:
+    """
+    Check consistency of column at the plant and utility level in two inputs.
+
+    Args:
+        df1: One dataframe to sum and check consistency with ``df2``. Both
+            dfs should have columns ``data_col`` as well as identifying
+            columns: ``['report_date', 'data_source', 'utility_id_pudl',
+            'plant_id_eia']``
+        df2: Other dataframe to sum and check consistency against ``df1``.
+        data_col: data column to check. Column must be in both ``df1`` and
+            ``df2``.
+
+    """
+    util_test = gb_test(
+        df1,
+        df2,
+        data_col=data_col,
+        by=['report_year', 'data_source', 'utility_id_pudl']
+    )
+
+    logger.info(
+        f"Duke utility level data is consistent for {data_col}!! Go forth and "
+        "close them coal plants lil tables")
+
+    plant_test = gb_test(
+        df1,
+        df2,
+        data_col=data_col,
+        by=['report_year', 'data_source', 'utility_id_pudl', 'plant_id_eia']
+    )
+
+    util_bad = util_test[~util_test.match & util_test.match.notnull()]
+    plant_bad = plant_test[~plant_test.match & plant_test.match.notnull()]
+    logger.warn(
+        f"We have {len(util_bad)} utilities and {len(plant_bad)} "
+        f"plants who's {data_col} doesn't match the input."
+    )
+
+    # we know the duke utilities should be correct!
+    assert(util_test.loc[2018, 'FERC', 90].match)
+    assert(util_test.loc[2018, 'FERC', 97].match)
+    return util_bad, plant_bad
+
+
+def gb_test(
+    df1: pd.DataFrame,
+    df2: pd.DataFrame,
+    data_col: str,
+    by: List[str]
+) -> pd.DataFrame:
+    """
+    Merge two grouped input tables to determine if summed data column matches.
+
+    Args:
+        df1: One dataframe to sum and check consistency with ``df2``.
+        df2: Other dataframe to sum and check consistency against ``df1``.
+        data_col: data column to check. Column must be in both ``df1`` and
+            ``df2``.
+    """
+    return (
+        pd.merge(
+            _group_sum_col(
+                df1, data_col=data_col, by=by),
+            _group_sum_col(
+                df2, data_col=data_col, by=by),
+            right_index=True, left_index=True,
+            suffixes=('_1', '_2'),
+            how='outer'
+        )
+        .assign(
+            match=lambda x: np.where(
+                x[f"{data_col}_1"].notnull() & x[f"{data_col}_2"].notnull(),
+                np.isclose(x[f"{data_col}_1"], x[f"{data_col}_2"]),
+                pd.NA
```

Comment:
What is the case where you end up assigning `pd.NA` here? Is that what happens when the column is null in one of the input dataframes?  What would happen if you ended up accidentally introducing extra null values in those input columns? Would the NA values be caught as failures?

Reply:
I changed the arguments for `np.isclose` to include `equal_nan=True`. this is what I was going for here: if both inputs are null, that is good and I want to return `True`.